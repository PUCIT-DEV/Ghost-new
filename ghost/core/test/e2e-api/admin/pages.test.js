const {agentProvider, fixtureManager, mockManager, matchers} = require('../../utils/e2e-framework');
const {anyArray, anyBoolean, anyContentVersion, anyEtag, anyLocationFor, anyObject, anyObjectId, anyISODateTime, anyString, anyUuid} = matchers;

const tierSnapshot = {
    id: anyObjectId,
    created_at: anyISODateTime,
    updated_at: anyISODateTime
};

const matchPageShallowIncludes = {
    id: anyObjectId,
    uuid: anyUuid,
    comment_id: anyString,
    url: anyString,
    authors: anyArray,
    primary_author: anyObject,
    tags: anyArray,
    primary_tag: anyObject,
    tiers: Array(2).fill(tierSnapshot),
    created_at: anyISODateTime,
    updated_at: anyISODateTime,
    published_at: anyISODateTime,
    post_revisions: anyArray,
    show_title_and_feature_image: anyBoolean
};

describe('Pages API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init('posts');
        await agent.loginAsOwner();
    });

    afterEach(function () {
        mockManager.restore();
    });

    describe('Update', function () {
        it('Can modify show_title_and_feature_image property', async function () {
            const page = {
                title: 'Test Page',
                status: 'draft'
            };

            const {body: pageBody} = await agent
                .post('/pages/?formats=mobiledoc,lexical,html', {
                    headers: {
                        'content-type': 'application/json'
                    }
                })
                .body({pages: [page]})
                .expectStatus(201);

            const [pageResponse] = pageBody.pages;

            await agent
                .put(`/pages/${pageResponse.id}/?formats=mobiledoc,lexical,html`)
                .body({
                    pages: [{
                        id: pageResponse.id,
                        show_title_and_feature_image: false, // default is true
                        updated_at: pageResponse.updated_at // satisfy collision detection
                    }]
                })
                .expectStatus(200)
                .matchBodySnapshot({
                    pages: [Object.assign({}, matchPageShallowIncludes, {
                        published_at: null,
                        show_title_and_feature_image: false
                    })]
                })
                .matchHeaderSnapshot({
                    'content-version': anyContentVersion,
                    etag: anyEtag,
                    'x-cache-invalidate': anyString
                });
        });
    });

    describe('Copy', function () {
        it('Can copy a page', async function () {
            const page = {
                title: 'Test Page',
                status: 'published'
            };

            const {body: pageBody} = await agent
                .post('/pages/?formats=mobiledoc,lexical,html', {
                    headers: {
                        'content-type': 'application/json'
                    }
                })
                .body({pages: [page]})
                .expectStatus(201)
                .matchBodySnapshot({
                    pages: [Object.assign({}, matchPageShallowIncludes)]
                });

            const [pageResponse] = pageBody.pages;

            await agent
                .post(`/pages/${pageResponse.id}/copy?formats=mobiledoc,lexical`)
                .expectStatus(201)
                .matchBodySnapshot({
                    pages: [Object.assign({}, matchPageShallowIncludes, {published_at: null})]
                })
                .matchHeaderSnapshot({
                    'content-version': anyContentVersion,
                    etag: anyEtag,
                    location: anyLocationFor('pages')
                });
        });
    });

    describe('Convert', function () {
        it('can convert a mobiledoc page to lexical', async function () { 
            const mobiledoc = JSON.stringify({
                version: '0.3.1',
                ghostVersion: '4.0',
                markups: [],
                atoms: [],
                cards: [],
                sections: [
                    [1, 'p', [
                        [0, [], 0, 'This is some great content.']
                    ]]
                ]
            });
            const expectedLexical = JSON.stringify({
                root: {
                    children: [
                        {
                            children: [
                                {
                                    detail: 0,
                                    format: 0,
                                    mode: 'normal',
                                    style: '',
                                    text: 'This is some great content.',
                                    type: 'text',
                                    version: 1
                                }
                            ],
                            direction: 'ltr',
                            format: '',
                            indent: 0,
                            type: 'paragraph',
                            version: 1
                        }
                    ],
                    direction: 'ltr',
                    format: '',
                    indent: 0,
                    type: 'root',
                    version: 1
                }
            });
            const pageData = {
                title: 'Test Post',
                status: 'published',
                mobiledoc: mobiledoc,
                lexical: null
            };

            const {body: pageBody} = await agent
                .post('/pages/?formats=mobiledoc,lexical,html', {
                    headers: {
                        'content-type': 'application/json'
                    }
                })
                .body({pages: [pageData]})
                .expectStatus(201);

            const [pageResponse] = pageBody.pages;

            await agent
                .put(`/pages/${pageResponse.id}/?formats=mobiledoc,lexical,html&convert_to_lexical=true`)
                .body({pages: [Object.assign({}, pageResponse)]})
                .expectStatus(200)
                .matchBodySnapshot({
                    pages: [Object.assign({}, matchPageShallowIncludes, {lexical: expectedLexical, mobiledoc: null})]
                })
                .matchHeaderSnapshot({
                    'content-version': anyContentVersion,
                    etag: anyEtag
                });
        });
    });
});
