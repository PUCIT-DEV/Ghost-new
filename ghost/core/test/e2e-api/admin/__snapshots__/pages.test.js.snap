// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages API Convert can convert a mobiledoc page to lexical 1: [body] 1`] = `
Object {
  "pages": Array [
    Object {
      "authors": Any<Array>,
      "canonical_url": null,
      "codeinjection_foot": null,
      "codeinjection_head": null,
      "comment_id": Any<String>,
      "count": Object {
        "negative_feedback": 0,
        "paid_conversions": 0,
        "positive_feedback": 0,
        "signups": 0,
      },
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "custom_excerpt": null,
      "custom_template": null,
      "excerpt": "This is some great content.",
      "feature_image": null,
      "feature_image_alt": null,
      "feature_image_caption": null,
      "featured": false,
      "frontmatter": null,
      "html": "<p>This is some great content.</p>",
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "lexical": "{\\"root\\":{\\"children\\":[{\\"children\\":[{\\"detail\\":0,\\"format\\":0,\\"mode\\":\\"normal\\",\\"style\\":\\"\\",\\"text\\":\\"This is some great content.\\",\\"type\\":\\"text\\",\\"version\\":1}],\\"direction\\":\\"ltr\\",\\"format\\":\\"\\",\\"indent\\":0,\\"type\\":\\"paragraph\\",\\"version\\":1}],\\"direction\\":\\"ltr\\",\\"format\\":\\"\\",\\"indent\\":0,\\"type\\":\\"root\\",\\"version\\":1}}",
      "meta_description": null,
      "meta_title": null,
      "mobiledoc": null,
      "og_description": null,
      "og_image": null,
      "og_title": null,
      "post_revisions": Any<Array>,
      "primary_author": Any<Object>,
      "primary_tag": Any<Object>,
      "published_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "reading_time": 0,
      "show_title_and_feature_image": Any<Boolean>,
      "slug": "test-post",
      "status": "published",
      "tags": Any<Array>,
      "tiers": Array [
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": null,
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": null,
          "monthly_price_id": null,
          "name": "Free",
          "slug": "free",
          "trial_days": 0,
          "type": "free",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": null,
          "yearly_price_id": null,
        },
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": "usd",
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": 500,
          "monthly_price_id": null,
          "name": "Default Product",
          "slug": "default-product",
          "trial_days": 0,
          "type": "paid",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": 5000,
          "yearly_price_id": null,
        },
      ],
      "title": "Test Post",
      "twitter_description": null,
      "twitter_image": null,
      "twitter_title": null,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "url": Any<String>,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "public",
    },
  ],
}
`;

exports[`Pages API Convert can convert a mobiledoc page to lexical 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "4013",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-cache-invalidate": "/*",
  "x-powered-by": "Express",
}
`;

exports[`Pages API Copy Can copy a page 1: [body] 1`] = `
Object {
  "pages": Array [
    Object {
      "authors": Any<Array>,
      "canonical_url": null,
      "codeinjection_foot": null,
      "codeinjection_head": null,
      "comment_id": Any<String>,
      "count": Object {
        "negative_feedback": 0,
        "paid_conversions": 0,
        "positive_feedback": 0,
        "signups": 0,
      },
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "custom_excerpt": null,
      "custom_template": null,
      "excerpt": null,
      "feature_image": null,
      "feature_image_alt": null,
      "feature_image_caption": null,
      "featured": false,
      "frontmatter": null,
      "html": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "lexical": null,
      "meta_description": null,
      "meta_title": null,
      "mobiledoc": "{\\"version\\":\\"0.3.1\\",\\"ghostVersion\\":\\"4.0\\",\\"markups\\":[],\\"atoms\\":[],\\"cards\\":[],\\"sections\\":[[1,\\"p\\",[[0,[],0,\\"\\"]]]]}",
      "og_description": null,
      "og_image": null,
      "og_title": null,
      "post_revisions": Any<Array>,
      "primary_author": Any<Object>,
      "primary_tag": Any<Object>,
      "published_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "show_title_and_feature_image": Any<Boolean>,
      "slug": "test-page-2",
      "status": "published",
      "tags": Any<Array>,
      "tiers": Array [
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": null,
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": null,
          "monthly_price_id": null,
          "name": "Free",
          "slug": "free",
          "trial_days": 0,
          "type": "free",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": null,
          "yearly_price_id": null,
        },
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": "usd",
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": 500,
          "monthly_price_id": null,
          "name": "Default Product",
          "slug": "default-product",
          "trial_days": 0,
          "type": "paid",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": 5000,
          "yearly_price_id": null,
        },
      ],
      "title": "Test Page",
      "twitter_description": null,
      "twitter_image": null,
      "twitter_title": null,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "url": Any<String>,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "public",
    },
  ],
}
`;

exports[`Pages API Copy Can copy a page 2: [body] 1`] = `
Object {
  "pages": Array [
    Object {
      "authors": Any<Array>,
      "canonical_url": null,
      "codeinjection_foot": null,
      "codeinjection_head": null,
      "comment_id": Any<String>,
      "count": Object {
        "negative_feedback": 0,
        "paid_conversions": 0,
        "positive_feedback": 0,
        "signups": 0,
      },
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "custom_excerpt": null,
      "custom_template": null,
      "excerpt": null,
      "feature_image": null,
      "feature_image_alt": null,
      "feature_image_caption": null,
      "featured": false,
      "frontmatter": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "lexical": null,
      "meta_description": null,
      "meta_title": null,
      "mobiledoc": "{\\"version\\":\\"0.3.1\\",\\"ghostVersion\\":\\"4.0\\",\\"markups\\":[],\\"atoms\\":[],\\"cards\\":[],\\"sections\\":[[1,\\"p\\",[[0,[],0,\\"\\"]]]]}",
      "og_description": null,
      "og_image": null,
      "og_title": null,
      "post_revisions": Any<Array>,
      "primary_author": Any<Object>,
      "primary_tag": Any<Object>,
      "published_at": null,
      "show_title_and_feature_image": Any<Boolean>,
      "slug": "test-page-copy",
      "status": "draft",
      "tags": Any<Array>,
      "tiers": Array [
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": null,
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": null,
          "monthly_price_id": null,
          "name": "Free",
          "slug": "free",
          "trial_days": 0,
          "type": "free",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": null,
          "yearly_price_id": null,
        },
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": "usd",
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": 500,
          "monthly_price_id": null,
          "name": "Default Product",
          "slug": "default-product",
          "trial_days": 0,
          "type": "paid",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": 5000,
          "yearly_price_id": null,
        },
      ],
      "title": "Test Page (Copy)",
      "twitter_description": null,
      "twitter_image": null,
      "twitter_title": null,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "url": Any<String>,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "public",
    },
  ],
}
`;

exports[`Pages API Copy Can copy a page 3: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "3727",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "location": StringMatching /https\\?:\\\\/\\\\/\\.\\*\\?\\\\/pages\\\\/\\[a-f0-9\\]\\{24\\}\\\\//,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Pages API Update Can modify hide_title_and_feature_image property 1: [body] 1`] = `
Object {
  "pages": Array [
    Object {
      "authors": Any<Array>,
      "canonical_url": null,
      "codeinjection_foot": null,
      "codeinjection_head": null,
      "comment_id": Any<String>,
      "count": Object {
        "negative_feedback": 0,
        "paid_conversions": 0,
        "positive_feedback": 0,
        "signups": 0,
      },
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "custom_excerpt": null,
      "custom_template": null,
      "excerpt": null,
      "feature_image": null,
      "feature_image_alt": null,
      "feature_image_caption": null,
      "featured": false,
      "frontmatter": null,
      "hide_title_and_feature_image": true,
      "html": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "lexical": null,
      "meta_description": null,
      "meta_title": null,
      "mobiledoc": "{\\"version\\":\\"0.3.1\\",\\"ghostVersion\\":\\"4.0\\",\\"markups\\":[],\\"atoms\\":[],\\"cards\\":[],\\"sections\\":[[1,\\"p\\",[[0,[],0,\\"\\"]]]]}",
      "og_description": null,
      "og_image": null,
      "og_title": null,
      "post_revisions": Any<Array>,
      "primary_author": Any<Object>,
      "primary_tag": Any<Object>,
      "published_at": null,
      "slug": "test-page",
      "status": "draft",
      "tags": Any<Array>,
      "tiers": Array [
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": null,
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": null,
          "monthly_price_id": null,
          "name": "Free",
          "slug": "free",
          "trial_days": 0,
          "type": "free",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": null,
          "yearly_price_id": null,
        },
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": "usd",
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": 500,
          "monthly_price_id": null,
          "name": "Default Product",
          "slug": "default-product",
          "trial_days": 0,
          "type": "paid",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": 5000,
          "yearly_price_id": null,
        },
      ],
      "title": "Test Page",
      "twitter_description": null,
      "twitter_image": null,
      "twitter_title": null,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "url": Any<String>,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "public",
    },
  ],
}
`;

exports[`Pages API Update Can modify hide_title_and_feature_image property 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "3667",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-cache-invalidate": "/p/80038ac3-4ec7-4f4b-8b30-9aacf306ee1e/",
  "x-powered-by": "Express",
}
`;

exports[`Pages API Update Can modify show_title_and_feature_image property 1: [body] 1`] = `
Object {
  "pages": Array [
    Object {
      "authors": Any<Array>,
      "canonical_url": null,
      "codeinjection_foot": null,
      "codeinjection_head": null,
      "comment_id": Any<String>,
      "count": Object {
        "negative_feedback": 0,
        "paid_conversions": 0,
        "positive_feedback": 0,
        "signups": 0,
      },
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "custom_excerpt": null,
      "custom_template": null,
      "excerpt": null,
      "feature_image": null,
      "feature_image_alt": null,
      "feature_image_caption": null,
      "featured": false,
      "frontmatter": null,
      "html": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "lexical": null,
      "meta_description": null,
      "meta_title": null,
      "mobiledoc": "{\\"version\\":\\"0.3.1\\",\\"ghostVersion\\":\\"4.0\\",\\"markups\\":[],\\"atoms\\":[],\\"cards\\":[],\\"sections\\":[[1,\\"p\\",[[0,[],0,\\"\\"]]]]}",
      "og_description": null,
      "og_image": null,
      "og_title": null,
      "post_revisions": Any<Array>,
      "primary_author": Any<Object>,
      "primary_tag": Any<Object>,
      "published_at": null,
      "show_title_and_feature_image": false,
      "slug": "test-page",
      "status": "draft",
      "tags": Any<Array>,
      "tiers": Array [
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": null,
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": null,
          "monthly_price_id": null,
          "name": "Free",
          "slug": "free",
          "trial_days": 0,
          "type": "free",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": null,
          "yearly_price_id": null,
        },
        Object {
          "active": true,
          "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "currency": "usd",
          "description": null,
          "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
          "monthly_price": 500,
          "monthly_price_id": null,
          "name": "Default Product",
          "slug": "default-product",
          "trial_days": 0,
          "type": "paid",
          "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
          "visibility": "public",
          "welcome_page_url": null,
          "yearly_price": 5000,
          "yearly_price_id": null,
        },
      ],
      "title": "Test Page",
      "twitter_description": null,
      "twitter_image": null,
      "twitter_title": null,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "url": Any<String>,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "public",
    },
  ],
}
`;

exports[`Pages API Update Can modify show_title_and_feature_image property 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "3728",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-cache-invalidate": Any<String>,
  "x-powered-by": "Express",
}
`;
